plugins/LilyPad/Config.cpp

```diff
@@ -2079,26 +2200,50 @@ INT_PTR CALLBACK GeneralDialogProc(HWND hWnd, unsigned int msg, WPARAM wParam, L
                         res--;
                         slot2 = res / 2;
                         port2 = res & 1;
+                        padtype2 = config.padConfigs[port2][slot2].type;
                         PadConfig padCfgTemp = config.padConfigs[port1][slot1];
                         config.padConfigs[port1][slot1] = config.padConfigs[port2][slot2];
                         config.padConfigs[port2][slot2] = padCfgTemp;
+                        if (config.swapPadtypes) {
+                            config.padConfigs[port2][slot2].type = config.padConfigs[port1][slot1].type;
+                            config.padConfigs[port1][slot1].type = padCfgTemp.type;
+                        }
                         for (int i = 0; i < dm->numDevices; i++) {
                             if (dm->devices[i]->type == IGNORE)
                                 continue;
-                            PadBindings bindings = dm->devices[i]->pads[port1][slot1];
-                            dm->devices[i]->pads[port1][slot1] = dm->devices[i]->pads[port2][slot2];
-                            dm->devices[i]->pads[port2][slot2] = bindings;
+                            if (config.swapPadtypes) {
```

Comment:
`swapPadTypes` isn't great naming if it's changing between individual binding swap and all bindings swap IMO.